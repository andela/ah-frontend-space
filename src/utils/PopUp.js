import React from 'react';
import '../assets/css/popup.scss';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as deleteArticleAction from '../actions/articleActions/deleteArticleActions';


export class Popup extends React.Component {
    deleteArticle = () => {
      console.log('deleting');
      const { slug, actions } = this.props;
      actions.deleteArticleAction(slug)
        .then(() => {
          window.location.replace('/');
        });
    };

    render() {
      const { closePopup } = this.props;
      return (
        <div className="popup">
          <div className="popup_inner">
            <p>Are you sure you want to delete?</p>
            <button type="button" className="btn-danger" onClick={this.deleteArticle}>Delete</button>
            <button type="button" className="btn-primary" onClick={closePopup}>Cancel</button>
          </div>
        </div>
      );
    }
}

export const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(deleteArticleAction, dispatch),
});

Popup.propTypes = {
  actions: PropTypes.shape().isRequired,
  slug: PropTypes.string.isRequired,
  closePopup: PropTypes.func.isRequired,
};

export default connect(
  null,
  mapDispatchToProps,
)(Popup);
