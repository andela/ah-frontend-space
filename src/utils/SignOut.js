
import { Component } from 'react';

export class SignOut extends Component {
  render() {
    sessionStorage.clear();
    return (
      window.location.replace('/')
    );
  }
}

export default SignOut;
