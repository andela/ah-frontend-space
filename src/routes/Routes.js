import React, { Component } from 'react';
import {
  Route, BrowserRouter, Switch,
} from 'react-router-dom';
import Home from '../components/HomePage/Home';
import SignInPage from '../containers/signIn/SignInPage';
import notFound from '../components/notfound';
import Profile from '../containers/profile/index';
import Hello from '../containers/Hello';
import NavBar from '../components/HomePage/Navbar';
import Footer from '../components/HomePage/Footer';
import SignupForm from '../containers/SignupForm';
import ConfirmEmailPage from '../containers/passwordReset/ConfirmEmailPage';
import ResetPasswordPage from '../containers/passwordReset/ResetPasswordPage';
import ArticleDetails from '../components/HomePage/HomeDetail';
import { SignOut } from '../utils/SignOut';

import editContainer from '../containers/profile/EditProfile';
import UpdateArticlePage from '../containers/article/UpdateArticlePage';

export class Routes extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <NavBar />
          <Footer />
          <Switch>
            <Route exact="true" path="/" component={Home} />
            <Route path="/signin" component={SignInPage} />
            <Route path="/hello" component={Hello} />
            <Route path="/signup" component={SignupForm} />
            <Route path="/confirm-email" component={ConfirmEmailPage} />
            <Route path="/reset-password" component={ResetPasswordPage} />
            <Route path="/signout" component={SignOut} />
            <Route exact path="/profile" component={Profile} />
            <Route path="/editprofile" component={editContainer} />
            <Route path="/edit-article" component={UpdateArticlePage} />
            <Route path="/:slug" component={ArticleDetails} />
            <Route component={notFound} />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

export default Routes;
