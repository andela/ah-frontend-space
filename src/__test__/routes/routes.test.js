import React from 'react';
import { shallow } from 'enzyme';
import Routes from '../../routes/Routes';
import { SignOut } from '../../utils/SignOut';


describe('<Routes/>', () => {
  it('should render Routes component', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call signout function when signout is clicked', () => {
    const wrapper = shallow(<SignOut />);
    expect(wrapper).toMatchSnapshot();
  });
});
