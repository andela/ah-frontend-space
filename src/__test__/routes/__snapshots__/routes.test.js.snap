// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes/> should call signout function when signout is clicked 1`] = `ShallowWrapper {}`;

exports[`<Routes/> should render Routes component 1`] = `ShallowWrapper {}`;
